wpimath/src/main/java/edu/wpi/first/math/autodiff/VariableBlock.java

```diff
@@ -0,0 +1,831 @@
+        result.set(i, j, sum);
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Matrix-scalar multiplication operator.
+   *
+   * @param rhs Operator right-hand side.
+   * @return Result of matrix-scalar multiplication.
+   */
+  public VariableMatrix times(double rhs) {
+    return times(new Variable(rhs));
+  }
+
+  /**
+   * Matrix-scalar multiplication operator.
+   *
+   * @param rhs Operator right-hand side.
+   * @return Result of matrix-scalar multiplication.
+   */
+  public VariableMatrix times(Variable rhs) {
+    var result = new VariableMatrix(rows(), cols());
+
+    for (int row = 0; row < result.rows(); ++row) {
+      for (int col = 0; col < result.cols(); ++col) {
+        result.set(row, col, get(row, col).times(rhs));
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Binary division operator.
+   *
+   * @param rhs Operator right-hand side.
+   * @return Result of division.
+   */
+  public VariableMatrix div(double rhs) {
+    return div(new Variable(rhs));
+  }
+
+  /**
+   * Binary division operator.
+   *
+   * @param rhs Operator right-hand side.
+   * @return Result of division.
+   */
+  public VariableMatrix div(Variable rhs) {
+    var result = new VariableMatrix(rows(), cols());
+
+    for (int row = 0; row < result.rows(); ++row) {
+      for (int col = 0; col < result.cols(); ++col) {
+        result.set(row, col, get(row, col).div(rhs));
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Binary addition operator.
+   *
+   * @param rhs Operator right-hand side.
+   * @return Result of addition.
+   */
+  public VariableMatrix plus(VariableMatrix rhs) {
+    assert rows() == rhs.rows() && cols() == rhs.cols();
+
+    var result = new VariableMatrix(rows(), cols());
+
+    for (int row = 0; row < result.rows(); ++row) {
+      for (int col = 0; col < result.cols(); ++col) {
+        result.set(row, col, get(row, col).plus(rhs.get(row, col)));
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Binary addition operator.
+   *
+   * @param rhs Operator right-hand side.
+   * @return Result of addition.
+   */
+  public VariableMatrix plus(VariableBlock rhs) {
+    assert rows() == rhs.rows() && cols() == rhs.cols();
+
+    var result = new VariableMatrix(rows(), cols());
+
+    for (int row = 0; row < result.rows(); ++row) {
+      for (int col = 0; col < result.cols(); ++col) {
+        result.set(row, col, get(row, col).plus(rhs.get(row, col)));
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Binary addition operator.
+   *
+   * @param rhs Operator right-hand side.
+   * @return Result of addition.
+   */
+  public VariableMatrix plus(SimpleMatrix rhs) {
+    assert rows() == rhs.getNumRows() && cols() == rhs.getNumCols();
+
+    var result = new VariableMatrix(rows(), cols());
+
+    for (int row = 0; row < result.rows(); ++row) {
+      for (int col = 0; col < result.cols(); ++col) {
+        result.set(row, col, get(row, col).plus(new Variable(rhs.get(row, col))));
```

Comment:
```suggestion
        result.set(row, col, get(row, col).plus(rhs.get(row, col)));
```
`.minus()` uses the `Variable.minus(double)` overload, so we might as well use the `Variable.plus(double)` overload here.
Ditto for `VariableMatrix.plus(SimpleMatrix)`.